tests/test_template.py

```
@@ -507,3 +510,139 @@ def test_generated_github_workflows_match_act_validation_contract(
+        "mutation": "mutation-mutmut.yml",
+        "mutation-rust": "mutation-cargo.yml",
+    }
+    for job_name, job in jobs.items():
+        uses = str(job.get("uses", "")) if isinstance(job, dict) else ""
+        workflow, separator, revision = uses.partition("@")
+        assert workflow == (
+            f"leynos/shared-actions/.github/workflows/{expected_workflows[job_name]}"
+        ), f"expected {job_name} to use its shared mutation workflow"
+        assert separator and re.fullmatch(r"[0-9a-f]{40}", revision), (
+            f"expected {job_name} to pin the shared mutation workflow to a commit SHA"
+        )
```

Ruff (0.15.21)

[warning] Assertion should be broken down into multiple parts

(PT018)